Update scr_marine_struct.gml

Blogaugis opened the pull request:
(Re)introduced stat randomization to hirelings,
Added more TODOs and other comments.

Blogaugis committed:
Update scr_add_man.gml

Added a "Combat Knife" to Skitarri soldiers;
Fixed "Inquisition Crusader" issue with gear item slot having an item belonging in a weapon group;
Added "Ork Armour" to "Ork Sniper"
Lots of TODO comments added.

Blogaugis committed:
Update scr_add_man.gml

Fixes the Crusaders not having Armour

Blogaugis committed:
Update scr_marine_struct.gml

(Re)introduced stat randomization to hirelings,
Added more TODO and other comments.

Blogaugis committed:
Light Bolter and Power Armour addition

Adding Light Bolter and Light Power armour, fixing the issue of SoBs having ranged weapon penalties.

Blogaugis committed:
Combat Knife addition

Skitarii receive a Combat Knife melee weapon.

EttyKitty reviewed (commented):
I'm okay with this PR. just a question of stat values and armour/weapon values.

Blogaugis committed:
Merge branch 'pr/14-hiresup2' into pr/14-hiresup1

Blogaugis committed:
Merge pull request #20 from Blogaugis/pr/14-hiresup1

Hireling fix merge

EttyKitty commented on scripts/scr_add_man/scr_add_man.gml:
```diff
@@ -35,12 +35,63 @@ function scr_add_man(man_role, target_company, spawn_exp, spawn_name, corruption
 			case "Skitarii":
 				obj_ini.wep1[target_company][good] = "Hellgun";
-				obj_ini.wep2[target_company][good] = ""; // Consider giving the poor fellow a "Combat Knife" or other melee weapon
+				obj_ini.wep2[target_company][good] = "Combat Knife"; // I should consider implementing a "Light Combat Knife", which is human-sized
```
I don't think this is necessary, as all units get a default melee weapon if they have none (think of rifle stocks, fists, claws, etc).
Additionally, the combat knife is like a human sized sword, it's not meant for small units.
And plus, mechanicus units often don't have any melee weapons together with ranged.

Blogaugis replied:
Ok, I'll remove the knife then.

Blogaugis replied:
And, knife is removed.

EttyKitty commented on scripts/scr_marine_struct/scr_marine_struct.gml:
```diff
@@ -523,148 +523,190 @@ global.base_stats = { //tempory stats subject to change by anyone that wishes to
+			strength : [12,1], // I don't think skitarii are particularly strong
+			constitution : [15,1],
+			weapon_skill : [12,1],
+			ballistic_skill : [20,1],			
+			dexterity : [25,1],
+			intelligence : [25,1],
+			wisdom : [10,1], // Hm, no, very unwise...
+			charisma : [5,1], // Talking in binary does not help to make many friends
 			religion : "cult_mechanicus",
-			piety : 20,
-			technology :30,
-			luck :5,
-			skills: {weapons:{"hellgun":1,}},	
-			start_gear:{"armour":"skitarii_armour", "wep1":"hellgun"},
+			piety : [20,1],
+			luck : 10, // I don't see the point to make them less lucky than space marines
+			technology : [30,1],
+			skills: {weapons:{"Hellgun":1, "Combat Knife":1}},
+			start_gear:{"armour":"Skitarii Armour", "wep1":"Hellgun", "wep2":"Combat Knife"},
 			base_group : "skitarii",
 	},
```
So, you've basically nerfed skitarii into the ground?

Blogaugis replied:
2.5 * 15 = 37.5. So, roughly 35 hp per skitarii.
Most hirelings are nerfed, as 10, 10, 10, 10 is or at least was the baseline for human stats.
Skitarii are stronger than guardsmen in this case.
I don't think it is that bad - their ranged stats are pretty alright.

Blogaugis replied:
Though, I should probably buff their armour a bit, right now it has like 5 AC only. Should I go ahead with it?

Blogaugis replied:
Although, I should probably lower the hands requirement for hellguns to just 1 as well, if I go at it.

Blogaugis replied:
Ok, their armour has slightly been buffed now, to compensate.
But the armour still probably needs more buffs.

EttyKitty replied:
My problem with these nerfs, is that you can justify them through lore perspective, but gameplay-vice you just made the unit much worse and kept the price the same, creating a disbalance.

Blogaugis replied:
Fair.
I have seen where the price for hirelings is located, assuming it works, I can lower it.
Should I also lower the price, then?

Also, assuming I am at it, should I also lower the disposition requirements for them? Since they are weaker now, anyway.

Blogaugis replied:
Addressed by lowering prices and disposition requirements. Marking it as resolved, then.

EttyKitty commented on scripts/scr_marine_struct/scr_marine_struct.gml:
```diff
@@ -523,148 +523,190 @@ global.base_stats = { //tempory stats subject to change by anyone that wishes to
-			charisma :2,
+			title : "Inquisition Crusader",
+			strength : [10,1], // 10 is considered standard for a chad-like Imperial Guardsman
+			constitution : [10,1],
```
And you've nerfed crusader hp.

Blogaugis replied:
Yeah, since they are regular humans, supposedly.
Or, are we going to treat them as something else? We probably need stat-buffing (constitution mostly) equipment.

EttyKitty commented on scripts/scr_marine_struct/scr_marine_struct.gml:
```diff
@@ -523,148 +523,190 @@ global.base_stats = { //tempory stats subject to change by anyone that wishes to
-			wisdom:5,
-			charisma :2,
+			strength : [10,1],
+			constitution : [10,1], // TODO - consider making it that hireling armour boosts constitution, and possibly other stats
```
And this as well.

Blogaugis replied:
Yes, sororitas are glorified guard. Unless we really want to buff regular humans?

EttyKitty reviewed (commented):
I'll be honest, aside from that knife, I don't have big issues with this pr.
Stat changes are hard to judge. It may just be weird to keep their costs the same but make them worse.

Blogaugis committed:
Tweaks 11-06

Removes combat knife for skitarii hireling, and adds Light Power Armor for crusader hireling instead of regular PA.

EttyKitty commented on scripts/scr_add_man/scr_add_man.gml:
```diff
@@ -91,13 +168,14 @@ function scr_add_man(man_role, target_company, spawn_exp, spawn_name, corruption
 				obj_ini.wep1[target_company][good] = "Sniper Rifle";
 				obj_ini.wep2[target_company][good] = "Choppa";
-				obj_ini.armour[target_company][good] = ""; // Consider giving "Ork Armour" to the fellow
+				obj_ini.armour[target_company][good] = "Ork Armour";
```
You gave Ork Armour here, but I think you're adding it only in #34.

Blogaugis committed:
More improvements/fixes 11-07

Added some code for Eldar and T'au hirelings,
some buffs added:
* lowering the amount of hands needed for some hireling weapons to 1.
* buffed skitarii armour, to compensate for lower stats overall.
Some nerfs:
* lowered starting EXP for some hirelings (sisters of battle/hospitalers)

EttyKitty commented on scripts/scr_weapon/scr_weapon.gml:
```diff
@@ -1681,6 +1542,166 @@ global.weapons = {
 	},
+// Hireling weapons
+	// Admech
+	"Hellgun": {
+		"abbreviation": "HllGun",
+		"attack": {
+			"standard": 30,
+			"master_crafted": 34,
+			"artifact": 38
+		},
+		"ammo": 10,
+		"range": 6.1,
+		"spli": 3,
+		"arp": 0,
+		"tags": [],
+		"ranged_hands": 1,
+		// "tags": ["arcane"], - how is it supposed to be arcane?
+		// ... (other attributes)
+	},
```
Adjusted Hellgun according to various sources and relation to our current weapons.
```suggestion
	"Hellgun": {
		"abbreviation": "HllGun",
		"attack": {
			"standard": 30,
			"master_crafted": 35,
			"artifact": 40
		},
		"description": "Known as a Hellgun or Hot-shot Lasgun, these high-energy weapons are more potent and destructive than common Lasguns, drawing more power for a more penetrative shot, but also more complex, requiring reinforced barrels, thermal-cooling cells and gyro-stabilized power packs.",
		"ammo": 20,
		"range": 12,
		"spli": 3,
		"arp": 0,
		"tags": ["las"],
		"melee_hands": 0,
		"ranged_hands": 1,
	},
```

EttyKitty commented on scripts/scr_weapon/scr_weapon.gml:
```diff
@@ -1681,6 +1542,166 @@ global.weapons = {
+		"tags": ["pistol", "las"],
+	},
+	// Other imperials
+	"Light Bolter": {
+		"abbreviation": "LiBltr",
+		"attack": {
+			"standard": 40,
+			"master_crafted": 45,
+			"artifact": 50
+		},
+		"description": "A smaller variant of Bolter, intended to be useable by unaugmented humans.",
+		"melee_hands": 0,
+		"ranged_hands": 1,
+		"ammo": 16,
+		"range": 12,
+		"spli": 3,
+		"arp": 0,
+		"tags": ["bolt"]
+	},
```
Nerfed this one a little bit more, again, according to various sources and our current weapons.
Also added Sarissa as a built-in weapon, as it's a bayonet.
```suggestion
	"Light Bolter": {
		"abbreviation": "LghtBltr",
		"attack": {
			"standard": 35,
			"master_crafted": 40,
			"artifact": 45
		},
		"description": "A smaller variant of Bolter, intended to be useable by unaugmented humans.",
		"melee_hands": 0,
		"ranged_hands": 1,
		"ammo": 16,
		"range": 10,
		"spli": 2,
		"arp": 0,
		"tags": ["bolt"],
		"second_profiles": ["Sarissa"]
	},
```

Blogaugis committed:
Update scripts/scr_weapon/scr_weapon.gml

Committing changes for hellgun.

Co-authored-by: EttyKitty <20323032+EttyKitty@users.noreply.github.com>

EttyKitty commented on scripts/scr_weapon/scr_weapon.gml:
```diff
@@ -1681,6 +1542,166 @@ global.weapons = {
+		"arp": 0,
+		"tags": ["bolt"]
+	},
+	"Sarissa": {
+		"abbreviation": "Saris",
+		"attack": {
+			"standard": 40,
+			"master_crafted": 45,
+			"artifact": 50
+		},
+		"description": "A vicious combat attachment that is attached to Bolters, in order to allow them to be used in melee combat.",
+		"melee_hands": 0,
+		"ranged_hands": 0,
+		"ammo": 0,
+		"range": 1,
+		"spli": 2,
+		"arp": 0,
+		"tags": ["sword"],
+	},
```
Nerfed Sarissa to be on pair with astartes Combat Knife, as this makes the most sense.
It's also mentioned that this is used as an axe and not sword.
```suggestion
	"Sarissa": {
		"abbreviation": "Saris",
		"attack": {
			"standard": 25,
			"master_crafted": 30,
			"artifact": 35
		},
		"description": "A vicious combat attachment that is attached to Bolters, in order to allow them to be used in melee combat.",
		"melee_hands": 0,
		"ranged_hands": 0,
		"ammo": 0,
		"range": 1,
		"spli": 1,
		"arp": 0,
		"tags": ["axe"],
	},
```

Blogaugis committed:
Update scripts/scr_weapon/scr_weapon.gml

Committing Light Bolter Changes.

Co-authored-by: EttyKitty <20323032+EttyKitty@users.noreply.github.com>

Blogaugis committed:
Update scripts/scr_weapon/scr_weapon.gml

Committing Sarissa Changes.

Co-authored-by: EttyKitty <20323032+EttyKitty@users.noreply.github.com>

EttyKitty commented on scripts/scr_weapon/scr_weapon.gml:
```diff
@@ -1681,6 +1542,167 @@ global.weapons = {
+		"arp": 0,
+		"tags": ["axe"],
+	},
+	"Hellrifle": {
+		"description": "Normally used by Radical Inquisitors, it appears an antiquated rifle but fires razor-sharp shards of Daemonic matter.",
+		"abbreviation": "HllRifle",
+		"attack": {
+			"standard": 150,
+			"master_crafted": 160,
+			"artifact": 170
+		},
+		"ammo": 10,
+		"range": 6.1,
+		"spli": 3,
+		"arp": 0,
+		"tags": ["rifle", "arcane"],
+		"ranged_hands": 1,
+		// ... (other attributes)
+	},
```
Updated description for this rifle, buffed damage and range, but reduced spli.
```suggestion
	"Hellrifle": {
		"description": "Extremely effective and intricate weapons frowned upon by more Puritanical Inquisitors. Fire shards of an unknown material. Speculated to be daemonic
in origin.",
		"abbreviation": "HllRifle",
		"attack": {
			"standard": 200,
			"master_crafted": 250,
			"artifact": 300
		},
		"ammo": 10,
		"range": 20,
		"spli": 0,
		"arp": 0,
		"tags": ["rifle", "arcane"],
		"ranged_hands": 1,
	},
```

Blogaugis replied:
I think it needs to have spli at least 1.
I remember back then, having spli as 0 meant weapons would not fire.

Blogaugis replied:
But, let's see anyway, maybe it has been fixed since then...

EttyKitty replied:
> I think it needs to have spli at least 1. I remember back then, having spli as 0 meant weapons would not fire.

If so, then Stalker Bolter also has 0 spli. The code seems to either ignore values below 1 or reset them to 1.
Either way, you can fix this on all other weapons in this file, as I can only modify stuff that you've modified before.

EttyKitty commented on scripts/scr_weapon/scr_weapon.gml:
```diff
@@ -2075,6 +2060,87 @@ global.gear = {
+				"artifact": 15 // Augmented
+			},
+		},
+		"Dragon Scales": {
+			"abbreviation": "DrgnArm",
+			"description": "Dragon Scales are an advanced armour ulitized by tech priests, it is remarkably lightweight for the protection it affords and is often greatly modified by it's wearer while also being designed to directly interface with the user's cybernetic body.",
+			"armour_value": {
+				"standard": 12, // Might want to buff this
+				"master_crafted": 14, // Augmented
+				"artifact": 16 // Augmented
+			}, // Considering the description, it should probably give other buffs, like extra hands and/or stats
+		},
```
Buffed this, according to DH rulebook, added a proper tag.
```suggestion
		"Dragon Scales": {
			"abbreviation": "DrgnScl",
			"description": "Dragon Scales are an advanced armour utilized by tech priests, it is remarkably lightweight for the protection it affords and is often greatly modified by it's wearer while also being designed to directly interface with the user's cybernetic body.",
			"armour_value": {
				"standard": 16,
				"master_crafted": 18,
				"artifact": 20
			},
			"melee_mod": {
				"standard": 0,
				"master_crafted": 5,
				"artifact": 10
			},
			"tags": ["power_armour"],
		},
```

EttyKitty commented on scripts/scr_weapon/scr_weapon.gml:
```diff
@@ -2075,6 +2060,87 @@ global.gear = {
+			"abbreviation": "DrgnArm",
+			"description": "Dragon Scales are an advanced armour ulitized by tech priests, it is remarkably lightweight for the protection it affords and is often greatly modified by it's wearer while also being designed to directly interface with the user's cybernetic body.",
+			"armour_value": {
+				"standard": 12, // Might want to buff this
+				"master_crafted": 14, // Augmented
+				"artifact": 16 // Augmented
+			}, // Considering the description, it should probably give other buffs, like extra hands and/or stats
+		},
+	// Sororitas and other imperials
+		"Light Power Armour": { // According to lore, lack of black carapace makes regular humans less capable to use PAs properly, but this is a concern for the future
+			"abbreviation": "LPArm",
+			"armour_value": {
+				"standard": 20,
+				"master_crafted": 25,
+				"artifact": 30
+			},
+			"ranged_mod": {
+				"standard": 0,
+				"master_crafted": 5,
+				"artifact": 10
+			},
+			"melee_mod": {
+				"standard": 0,
+				"master_crafted": 5,
+				"artifact": 10
+			},
+			"description": "A suit of light power armour, intended to be useable by the regular humans.",
+			"tags": ["power_armour"],
+		},
```
Nerfed this to comply with our current armour values. Why would you make it so strong?
```suggestion
		"Light Power Armour": { // According to lore, lack of black carapace makes regular humans less capable to use PAs properly, but this is a concern for the future
			"abbreviation": "LPwrArm",
			"armour_value": {
				"standard": 14,
				"master_crafted": 16,
				"artifact": 18
			},
			"ranged_mod": {
				"standard": 0,
				"master_crafted": 5,
				"artifact": 10
			},
			"melee_mod": {
				"standard": 0,
				"master_crafted": 5,
				"artifact": 10
			},
			"description": "A suit of light power armour, intended to be useable by the regular humans.",
			"tags": ["power_armour"],
		},
```

Blogaugis replied:
Because the hireling HP is very low, with what constitution values they start with.

Blogaugis replied:
But fine, Let's get this merged.

Blogaugis replied:
We also will need to nerf the armour stats in the hostile forces section.
To at least ensure that they are the same.

EttyKitty replied:
> Because the hireling HP is very low, with what constitution values they start with.

Cool? How is that a good reason to set armour values on light power armour higher than space marine MK7 power armour?

Blogaugis committed:
Update scripts/scr_weapon/scr_weapon.gml

Committing the Dragon Scales tweaks.

Co-authored-by: EttyKitty <20323032+EttyKitty@users.noreply.github.com>

Blogaugis committed:
Update scripts/scr_weapon/scr_weapon.gml

Committing Light Power Armour Changes.

Co-authored-by: EttyKitty <20323032+EttyKitty@users.noreply.github.com>

Blogaugis committed:
Update scripts/scr_weapon/scr_weapon.gml

Committing Hellrifle changes.

Co-authored-by: EttyKitty <20323032+EttyKitty@users.noreply.github.com>

EttyKitty commented on scripts/scr_add_man/scr_add_man.gml:
```diff
@@ -56,48 +107,75 @@ function scr_add_man(man_role, target_company, spawn_exp, spawn_name, corruption
+			case "Sanctioned Psyker":
+				obj_ini.wep1[target_company][good] = "Force Staff"; TODO - consider adding regular human version
+				obj_ini.wep2[target_company][good] = "Laspistol";
+				obj_ini.armuor[target_company][good] = "Power Armour"; TODO - consider adding imperial armour
+				obj_ini.gear[target_company][good] = "Psychic Hood"; TODO - consider adding regular human version
+				obj_ini.experience[target_company][good] = 50;
+				obj_ini.race[target_company][good] = 4;
+				unit = new TTRPG_stats("inquisition", target_company, good, "sanctioned_psyker"); TODO - add proper stat lines
+				break;
+			*/
+		// Faction 4.5 - Radical inquisitors, perhaps?
 		// Faction 5 - Sisters of Battle
 			case "Sister of Battle":
-				obj_ini.wep1[target_company][good] = "Bolter"; // Might want to create a "Light Bolter" variant for this one
+				obj_ini.wep1[target_company][good] = "Light Bolter";
 				obj_ini.wep2[target_company][good] = "Sarissa";
-				obj_ini.armour[target_company][good] = "Power Armour"; // Same here, Sororitas are glorified guard
-				obj_ini.experience[target_company][good] = 60;
+				obj_ini.armour[target_company][good] = "Light Power Armour";
+				obj_ini.experience[target_company][good] = 20;
 				obj_ini.race[target_company][good] = 5;
 				unit = new TTRPG_stats("adeptus_sororitas", target_company, good, "sister_of_battle");
 				break;
 			case "Sister Hospitaler":
-				obj_ini.wep1[target_company][good] = "Bolter"; // Same here
+				obj_ini.wep1[target_company][good] = "Light Bolter";
 				obj_ini.wep2[target_company][good] = "Sarissa";
-				obj_ini.armour[target_company][good] = "Power Armour"; // Same here
-				obj_ini.experience[target_company][good] = 100;
+				obj_ini.armour[target_company][good] = "Light Power Armour";
+				obj_ini.experience[target_company][good] = 50;
 				obj_ini.gear[target_company][good] = "Sororitas Medkit";
 				obj_ini.race[target_company][good] = 5;
 				unit = new TTRPG_stats("adeptus_sororitas", target_company, good, "sister_hospitaler");
 				break;
+			/* TODO - if we get to recruiting allied or other leaders, lot of stuff needs to be implemented
+			case "Prioress":
+				obj_ini.wep1[target_company][good] = "Bolter"; TODO - give a cool weapon
+				obj_ini.wep2[target_company][good] = "Sarissa"; TODO - same
+				obj_ini.armour[target_company][good] = "Power Armour"; TODO - Give cool armour
```
Can remove Sarissa from them, as it's now built-in into the bolter.
```suggestion
			case "Sister of Battle":
				obj_ini.wep1[target_company][good] = "Light Bolter";
				obj_ini.wep2[target_company][good] = "";
				obj_ini.armour[target_company][good] = "Light Power Armour";
				obj_ini.experience[target_company][good] = 20;
				obj_ini.race[target_company][good] = 5;
				unit = new TTRPG_stats("adeptus_sororitas", target_company, good, "sister_of_battle");
				break;
			case "Sister Hospitaler":
				obj_ini.wep1[target_company][good] = "Light Bolter";
				obj_ini.wep2[target_company][good] = "";
				obj_ini.armour[target_company][good] = "Light Power Armour";
				obj_ini.experience[target_company][good] = 50;
				obj_ini.gear[target_company][good] = "Sororitas Medkit";
				obj_ini.race[target_company][good] = 5;
				unit = new TTRPG_stats("adeptus_sororitas", target_company, good, "sister_hospitaler");
				break;
			/* TODO - if we get to recruiting allied or other leaders, lot of stuff needs to be implemented
			case "Prioress":
				obj_ini.wep1[target_company][good] = "Light Bolter"; TODO - give a cool weapon
				obj_ini.wep2[target_company][good] = ""; TODO - same
```

Blogaugis replied:
Ok, this should be fixed now.

EttyKitty commented on scripts/scr_marine_struct/scr_marine_struct.gml:
```diff
@@ -523,148 +523,190 @@ global.base_stats = { //tempory stats subject to change by anyone that wishes to
+				weapons:{
+					"Light Bolter":1, "Bolt Pistol":1, "Chainsword":1, "Sarissa":1}},
+			start_gear:{"armour":"Light Power Armour", "wep1":"Light Bolter", "wep2":"Sarissa"},
 			base_group : "human",
```
Can also remove it from here, as it's in the bolter.
```suggestion
			start_gear:{"armour":"Light Power Armour", "wep1":"Light Bolter", "wep2":""},
```

EttyKitty commented on scripts/scr_marine_struct/scr_marine_struct.gml:
```diff
@@ -523,148 +523,190 @@ global.base_stats = { //tempory stats subject to change by anyone that wishes to
+				weapons:{
+					"Light Bolter":2, "Bolt Pistol":2, "Chainsword":2, "Sarissa":2}},
+			start_gear:{"armour":"Light Power Armour", "wep1":"Light Bolter", "wep2":"Sarissa", "gear":"Sororitas Medkit"},
 			base_group : "human",
```
And here:
```suggestion
			start_gear:{"armour":"Light Power Armour", "wep1":"Light Bolter", "wep2":"", "gear":"Sororitas Medkit"},
```

Blogaugis committed:
Update scripts/scr_marine_struct/scr_marine_struct.gml

Removing sarissa.

Co-authored-by: EttyKitty <20323032+EttyKitty@users.noreply.github.com>

Blogaugis committed:
Update scripts/scr_marine_struct/scr_marine_struct.gml

removing sarissa

Co-authored-by: EttyKitty <20323032+EttyKitty@users.noreply.github.com>

Blogaugis committed:
Update scr_weapon.gml

Giving all the weapons `spli` value.

Blogaugis committed:
Merge branch 'pr/14-hiresup2' of https://github.com/Blogaugis/Space-Brigade-Master into pr/14-hiresup2

Blogaugis committed:
Update scr_add_man.gml

Removes Sarissas.

Blogaugis committed:
Update scr_marine_struct.gml

removes sarissas and adds missing code bits.

Blogaugis committed:
Price and disposition requirement tweaks

Lower prices and disposition requirements for imperial factions:
* ecclesiarchy: 40 and 50 requisition, 40 and 45 disposition requirement;
* inquisition: 20 requisition, 25 disposition;
* admech: 25 and 55 disposition; 15 requisition.

EttyKitty commented on scripts/scr_marine_struct/scr_marine_struct.gml:
```diff
@@ -523,148 +523,190 @@ global.base_stats = { //tempory stats subject to change by anyone that wishes to
+			luck : 10,
+			technology : [55,3],
+			skills: {weapons:{"power_weapon":2,}}, // Does this even work?
+			start_gear:{"armour":"dragon_scales", "wep1":"power_weapon"}, // Gear items are not mentioned here
```
You may want to fix names here, like you did with Skitarii.

Blogaugis replied:
Ok...

Blogaugis replied:
I hope "mobi" actually works...

Blogaugis committed:
Update scr_marine_struct.gml

Fixing names for techpriest.

Blogaugis committed:
Update scr_weapon.gml

Fixing compile error.

Blogaugis commented:
Apparently, it was implemented in 0.10.0 already.
